### ISSUE TITLE
FR: support ImageOverlay

### ISSUE THREAD

**Issue by name-name** [rstudio/leaflet#73]
Hi is it possible or planned to support leaflet's ImageOverlay: http://leafletjs.com/reference.html#imageoverlay

Specifically I'd like to add a transparent plot generated by R to the map (some splines through the markers), which should be zoomed/panned etc when the map is.

**Comment by jcheng5** [comment]
I'm planning on implementing that this week. Do you have a code sample for how you're generating the image?

**Comment by name-name** [comment]
Well, creating a spline curve in 2D with xspline for a fixed set of points is explained here:
http://stackoverflow.com/questions/28093925/calculate-a-2d-spline-curve-in-r

However, I'm not quite sure if this works at all on a zoomable map:
1) I probably need to project coordinates to the 2D points then create the plot and calculate the image bounds as lat/long (i.e. reverse projection)
2) not sure if having a fixed size image with given "imageBounds" (in lat/long) is enough when we zoom - but looking at a VERY simple (right-angled) triangle on a map: zooming doesn't change the angles!

Tested with this:

        ## app.R ##
        library(shiny)
        library(shinydashboard)
        library(leaflet)
        
        lng = c(0,0,10,0)
        lat = c(0,10,0,0)
        triangle = data.frame(lat,lng)
        
        
        runApp(list(
          ui = dashboardPage(
            dashboardHeader(title = ""),
            dashboardSidebar(disable=T),
            dashboardBody(
                leafletOutput("map", height = 600)
            )
          ),
          server = function(input, output) {
            
            output$map <- renderLeaflet({
              
              map <- leaflet() %>% addTiles()
              
              map %>% addPolylines(data=triangle, fillOpacity = 1.0,
                                   lat = ~lat, lng = ~lng)  # :-( 
            })
          }
        ))

**Comment by jcheng5** [comment]
> I probably need to project coordinates to the 2D points then create the plot and calculate the image bounds as lat/long (i.e. reverse projection)

This kind of thing is why I wanted a code sample--I wanted to know whether you are generating a projected or unprojected image.

And pardon my ignorance, but why would you expect the angles to change as you zoom?

**Comment by name-name** [comment]
I added the leaflet link in my comment above. Either IProjection is available in the leaflet package (maybe another FR^^) or I code it myself in R - shouldn't be that hard:
https://github.com/Leaflet/Leaflet/tree/master/src/geo/projection

And no, I just started thinking about angles and projections (and resolution), so it was just a quick test for me to see if that could work at all.
Thanks for your quick support!
